crates/router/src/core/payment_methods/cards.rs

```diff
@@ -383,7 +383,7 @@ pub async fn list_payment_methods(
             error.to_not_found_response(errors::ApiErrorResponse::MerchantAccountNotFound)
         })?;
 
-    let mut response: HashSet<api::ListPaymentMethod> = HashSet::new();
+    let mut response: HashMap<api::ListPaymentMethod, Vec<String>> = HashMap::new();
```

Comment:
`api::ListPaymentMethod` is a struct, Here we are keeping the struct as the key of the hash map.

Keeping this big struct as key don't make sense.